### PR TITLE
JDK-8262165: NMT report should state how many callsites had been skipped

### PR DESCRIPTION
**Pull request by tstuefe**
(This is a breakout from JDK-8261238, which seems to be stuck in review (https://github.com/openjdk/jdk/pull/2428). I factor out the display changes, which hopefully are non-controversial enough to get upstream easily.)

NMT silently omits call sites in its report whose allocation rate is too small to get over certain, hidden, thresholds:
a) there is a hard wired threshold of 1K in the baseline collector
b) we also skip all sites whose size is less than what the unit is (eg scale=G, we skip all allocations < 1G

NMT report should state how many call sites had been omitted.

----

The patch adds a counter in the NMT reporter to count omitted call sites, then prints that counter out. 

In base report, just a generic message is shown to notify the user that categories > scale are ignored:

```
thomas@mainframe$ jcmd Interl VM.native_memory scale=m
2847:

Native Memory Tracking:

(Omitting categories weighting less than 1MB)

Total: reserved=5540MB, committed=147MB
-                 Java Heap (reserved=3996MB, committed=34MB)
                            (mmap: reserved=3996MB, committed=34MB) 
<skip>
```

In detail report, the total count of omitted call sites is displayed at the bottom:

```
thomas@mainframe$ jcmd Interl VM.native_memory scale=m detail
2847:

<skip>

[0x00007f4e062e3e6b] os::reserve_memory(unsigned long, bool, MEMFLAGS)+0x16b
[0x00007f4e05ac4f47] MmapArrayAllocator<unsigned char>::allocate(unsigned long, MEMFLAGS)+0x127
[0x00007f4e05ac4cb2] G1FromCardCache::initialize(unsigned int)+0xe2
[0x00007f4e05a863f1] G1CollectedHeap::initialize()+0x3c1
                             (reserved=1MB, committed=1MB Type=GC)

(183 call sites weighting less than 1MB each omitted.)

```

Note: this patch only takes care of the sites omitted due to the scale threshold (b). It does not count how many sites had been omitted due to the baseline threshold (a). Since I still hope to remove that threshold completely with JDK-8261238, I did not put any work toward modifying the output here.

----

Tests: I manually ran runtime/NMT tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262165](https://bugs.openjdk.java.net/browse/JDK-8262165): NMT report should state how many callsites had been skipped

### Reviewers
 * [Zhengyu Gu](https://openjdk.java.net/census#zgu) (@zhengyu123 - **Reviewer**) ⚠️ Review applies to a6bbf78a84a8add3bbbafb815e589e7590cbde68
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to a6bbf78a84a8add3bbbafb815e589e7590cbde68

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/2687/head:pull/2687` \
`$ git checkout pull/2687`

Update a local copy of the PR: \
`$ git checkout pull/2687` \
`$ git pull https://git.openjdk.java.net/jdk pull/2687/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2687`

View PR using the GUI difftool: \
`$ git pr show -t 2687`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/2687.diff">https://git.openjdk.java.net/jdk/pull/2687.diff</a>

</details>

### CODE REVIEW

**Comment by tstuefe** [comment]
Friendly Ping.

**Review by zhengyu123 — APPROVED** [review]
Looks good to me

**Comment by tstuefe** [comment]
> Looks good to me

Thanks Zhengyu!

**Comment by tstuefe** [comment]
Can I have a second review please? Or could we, in this case, count my own pair of eyes as second reviewer, if no-one wants to look at NMT?

This change is really not that difficult.

Thanks, Thomas

File: src/hotspot/share/services/memReporter.cpp
Comment (shipilev) [inline src/hotspot/share/services/memReporter.cpp:307]: I think the Hotspot style is to say `num_omitted++`.

File: src/hotspot/share/services/memReporter.hpp
Comment (shipilev) [inline src/hotspot/share/services/memReporter.hpp:150]: Here and two lines below, comments do not need trailing points.

**Review by shipilev — APPROVED** [review]
Pretty simple, sure. Only minor nits.

**Comment by tstuefe** [comment]
> Pretty simple, sure. Only minor nits.

Great, thanks, Aleksey. Good to have this finally off the table. I'll add the suggested modifications before pushing.

**Pull request closed without merging**
